salt/modules/firewalld.py

```diff
@@ -629,8 +629,6 @@ def add_port(zone, port, permanent=True):
 
         salt '*' firewalld.add_port internal 443/tcp
     '''
-    if not get_masquerade(zone):
-        add_masquerade(zone)
```

Comment:
I don't know that we should just outright change this behavior.  We should probably have an argument passed that disables it.

It is possible that people are using this feature as is.

Reply:
As far as I can tell, this behavior is undocumented and seems to be included accidentally, as it doesn't look like it has anything to do with the `add_port` and `add_port_fwd` functions. From what I understand, the module is supposed to be a wrapper for `firewall-cmd`, the CLI for firewalld. The [firewall-cmd documentation](http://www.firewalld.org/documentation/man-pages/firewall-cmd.html) doesn't show that the `--add-port` or `--add-forward-port` flags should affect masquerading, so it seems incongruous with the expected behavior. Additionally, the `firewalld` execution module has a function `add_masquerade` that enables masquerading, so no functionality is removed by making this change.  
We could leave it in and add an argument to skip it, but at that point it feels like we're leaving in a bug and adding an argument to fix the bug.

Reply:
The problem is that it is in there, and changing it could change the behavior.

It should be another argument that defaults to True, but can have the default changed in 2 release version, so you can use the salt.utils.warn_until, and mark it for Neon, and we will go ahead and make the change once Fluoride release is out.

Reply:
When should `salt.utils.warn_until` be called? (i.e. any time the argument isn't specified?)

Reply:
do it basically like you did for the prune services.

set it to True, if None is passed, and then do the warn_until. there.